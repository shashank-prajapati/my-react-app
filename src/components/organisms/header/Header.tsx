import React, { useState } from "react";
import BlinkistLogo from "../../../media/blinkistLogo.png";
import Box, { BoxProps } from "@mui/material/Box";
import IconButton from "../../atoms/iconButton/IconButton";
import AvatarButton from "../../molecules/avatarButton/AvatarButton";
import Image from "../../atoms/image/Image";
import ButtonAtom from "../../atoms/button/ButtonAtom";
import { theme } from "../../../themes/themes";
import { useAuth0 } from "@auth0/auth0-react";
import { Menu, MenuItem } from "@mui/material";
import { ReactComponent as SearchIcon } from "../../../media/searchIcon.svg";

type Props = {
  btnClick?: React.MouseEventHandler<HTMLButtonElement>;
  btnIcon?: React.ReactNode;
  gotoMyLibrary?: any;
};

function Header(props: Props) {
  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
  const open = Boolean(anchorEl);
  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
    setAnchorEl(event.currentTarget);
  };
  const handleClose = () => {
    setAnchorEl(null);
  };
  const { btnClick, btnIcon, gotoMyLibrary } = props;
  const { loginWithRedirect, isAuthenticated, logout, user } = useAuth0();

  const logoutBtnClick = () => {
    handleClose();
    logout({ returnTo: window.location.origin });
  };

  return (
    <div>
      <Box
        sx={{
          display: "flex",
          alignItems: "center",
          px: "250px",
          py: "30px",
          justifyContent: "space-between",
          flexDirection: "row",
        }}
      >
        <Box
          sx={{
            display: "flex",
            alignItems: "center",
          }}
        >
          <Image source={BlinkistLogo} imgHeight="26px" imgWidth="124px" />
          <IconButton iconComponent={SearchIcon} sx={{ ml: 10 }} />
          <ButtonAtom
            ml={10}
            onClick={btnClick}
            endIcon={btnIcon}
            borderRadius={0}
            hoverBottomBorder={`1px solid ${theme.palette.primary.main}`}
            textColor="navTextColors.main"
            variant="text"
            hoverBackGroundColor="white"
          >
            Explore
          </ButtonAtom>
          <ButtonAtom
            variant="text"
            textColor="navTextColors.main"
            size="medium"
            onClick={() => gotoMyLibrary()}
            ml={10}
          >
            My Library
          </ButtonAtom>
        </Box>
        <Box
          sx={{
            display: "flex",
            alignItems: "center",
          }}
        >
          {!isAuthenticated && (
            <ButtonAtom
              variant="text"
              textColor="navTextColors.main"
              onClick={() => loginWithRedirect()}
            >
              Signin
            </ButtonAtom>
          )}

          {isAuthenticated && (
            <AvatarButton
              data-testid="account-Btn"
              onClick={handleClick}
              isAuthenticated={isAuthenticated}
              userName={user && user.email}
            />
          )}
          <Menu
            anchorOrigin={{
              vertical: "bottom",
              horizontal: "right",
            }}
            transformOrigin={{
              vertical: "top",
              horizontal: "right",
            }}
            id="basic-menu"
            anchorEl={anchorEl}
            open={open}
            onClose={handleClose}
            MenuListProps={{
              "aria-labelledby": "basic-button",
            }}
          >
            <MenuItem onClick={logoutBtnClick}>Logout</MenuItem>
          </Menu>
        </Box>
      </Box>
    </div>
  );
}

export default Header;
