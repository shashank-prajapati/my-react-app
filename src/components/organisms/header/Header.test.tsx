import * as React from "react";
import { render, fireEvent, screen } from "@testing-library/react";
import Header from "./Header";

describe("Testing Explore Menu", () => {
  it("should match the Explore Menu snapshot", () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });

  it("should call the on click function", () => {
    const handleClick = jest.fn();
    const { container } = render(<Header btnClick={handleClick} />);

    fireEvent.click(screen.getByText("Explore"));
  });
});
