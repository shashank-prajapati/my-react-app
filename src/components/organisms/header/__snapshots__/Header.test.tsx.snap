// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Header should match the Header snapshot 1`] = `
<div>
  <div>
    <div
      class="MuiBox-root css-1ggpog0"
    >
      <div
        class="MuiBox-root css-70qvj9"
      >
        <img
          alt="Book"
          height="26px"
          src="blinkistLogo.png"
          style="display: block;"
          width="124px"
        />
        <div>
          <button
            aria-label="search button"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1ig5hch-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-lw6gi8-MuiSvgIcon-root"
              focusable="false"
            >
              searchIcon.svg
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-q987oy-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Explore
          <span
            class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-hpr88x-MuiSvgIcon-root"
              focusable="false"
            >
              expandMore.svg
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-72xl0z-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          My Library
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiBox-root css-70qvj9"
      >
        <button
          class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-yw3ykx-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Signin
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
